### PR TITLE
QA

### PR DESCRIPTION
**Pull request by a-belhadj**

### CODE REVIEW

File: Squest/settings.py [hunk 1]
```diff
@@ -135,14 +137,14 @@
...
-        'PORT': DB_PORT,
-    }
+    # 'psql': {
+    #     'ENGINE': "django.db.backends.postgresql",
+    #     'NAME': DB_DATABASE,
+    #     'USER': DB_USER,
+    #     'PASSWORD': DB_PASSWORD,
+    #     'HOST': DB_HOST,
+    #     'PORT': DB_PORT,
+    # }
```
Comment (Sispheor) [inline Squest/settings.py:147]: Our current process doesn't propose to update settings directly. We use env variables. So this should be wraped by a condition that check if we are using postgress.

File: Squest/utils/squest_filter.py [hunk 2]
```diff
@@ -1,18 +1,38 @@
+import django_filters
 from django_filters import FilterSet
 from django.forms import TextInput, Select, SelectMultiple, CheckboxInput
 
 
 class SquestFilter(FilterSet):
+    per_page = django_filters.NumberFilter(
```
Comment (EliasBoulharts) [inline Squest/utils/squest_filter.py:7]: import NumberFilter only and remove `import django_filters` bellow

File: service_catalog/filters/global_hook_filter.py [hunk 3]
```diff
@@ -1,46 +1,14 @@
...
-        widget=SelectMultiple())
-
-    state_instance = MultipleChoiceFilter(
-        label="Instance state",
-        method='add_states_in_filter',
-        choices=InstanceState.choices,
-        null_value=None,
-        widget=SelectMultiple(attrs={'data-live-search': "true"}))
+        model = InstanceHook
+        fields = ['name', 'state']
```
Comment (EliasBoulharts) [inline service_catalog/filters/global_hook_filter.py:8]: add operation and service ?

File: service_catalog/filters/global_hook_filter.py [hunk 4]
```diff
@@ -1,46 +1,14 @@
...
-
-    def filter_queryset(self, queryset):
-        queryset = super(GlobalHookFilter, self).filter_queryset(queryset)
-        tmp = self.state_filter_value
-        self.state_filter_value = []
-        return queryset.filter(state__in=tmp)
+class RequestHookFilter(SquestFilter):
+    class Meta:
+        model = RequestHook
+        fields = ['name', 'state']
```
Comment (EliasBoulharts) [inline service_catalog/filters/global_hook_filter.py:14]: Same that above?

File: service_catalog/forms/global_hook_forms.py [hunk 5]
```diff
@@ -1,38 +1,14 @@
...
-                                           widget=forms.Select(attrs={'class': 'form-control'}))
 
-    def clean(self):
-        cleaned_data = super(GlobalHookForm, self).clean()
-        model = cleaned_data.get('model')
-        state = cleaned_data.get('state')
+class InstanceHookForm(SquestModelForm):
+    class Meta:
+        model = InstanceHook
+        fields = ["name", "service", "state", "job_template", "extra_vars"]
```
Comment (EliasBoulharts) [inline service_catalog/forms/global_hook_forms.py:8]: add operation

File: service_catalog/forms/global_hook_forms.py [hunk 6]
```diff
@@ -1,38 +1,14 @@
...
-                'state': f"'{state}' is not a valid state of model '{model}'"
-            })
-        return cleaned_data
 
+class RequestHookForm(SquestModelForm):
     class Meta:
-        model = GlobalHook
-        fields = ["name", "model", "service", "operation", "state", "job_template", "extra_vars"]
+        model = RequestHook
+        fields = ["name", "operation", "state", "job_template", "extra_vars"]
```
Comment (EliasBoulharts) [inline service_catalog/forms/global_hook_forms.py:14]: add service

File: service_catalog/models/instance.py [hunk 7]
```diff
@@ -201,21 +201,25 @@ def reset_to_last_stable_state(self):
...
-        """
-        HookManager.trigger_hook(sender, instance, name, source, target, *args, **kwargs)
-
     @classmethod
     def on_create_call_hook_manager(cls, sender, instance, created, *args, **kwargs):
         if created:
             HookManager.trigger_hook(sender=sender, instance=instance, name="create", source="create",
                                      target=InstanceState.PENDING, *args, **kwargs)
 
+    @classmethod
```
Comment (EliasBoulharts) [inline service_catalog/models/instance.py:210]: presave?

File: service_catalog/models/instance.py [hunk 8]
```diff
@@ -201,21 +201,25 @@ def reset_to_last_stable_state(self):
...
+    def on_change(cls, sender, instance, *args, **kwargs):
+        if instance.id is not None:
+            previous = Instance.objects.get(id=instance.id)
+            if previous.state != instance.state:
+                HookManager.trigger_hook(sender=sender, instance=instance, name="on_change_instance", source=previous.state, target=instance.state,
+                                         *args, **kwargs)
```
Comment (EliasBoulharts) [inline service_catalog/models/instance.py:216]: Not run on create

File: service_catalog/models/request.py [hunk 9]
```diff
@@ -392,9 +384,14 @@ def on_create(cls, sender, instance, created, *args, **kwargs):
 
     @classmethod
     def on_change(cls, sender, instance, *args, **kwargs):
-        # reset all approval step to pending if an approval workflow was attached to the request
         if instance.id is not None:
             previous = Request.objects.get(id=instance.id)
+            if previous.state != instance.state:
+                HookManager.trigger_hook(sender=sender, instance=instance, name="on_change_request",
+                                         source=previous.state, target=instance.state,
+                                         *args, **kwargs)
```
Comment (EliasBoulharts) [inline service_catalog/models/request.py:392]: Not run on create

File: Squest/settings.py [hunk 10]
```diff
@@ -42,7 +37,7 @@
 DATE_FORMAT = os.environ.get('DATE_FORMAT', "%d %b, %Y  %H:%M")
 LOGIN_HELPER_TEXT = os.environ.get('LOGIN_HELPER_TEXT', None)
 IS_DEV_SERVER = str_to_bool(os.environ.get('IS_DEV_SERVER', False))
-
+SQL_DEBUG = str_to_bool(os.environ.get('SQL_DEBUG', False))
```
Comment (Sispheor) [inline Squest/settings.py:40]: add to doc or remove

File: Squest/settings.py [hunk 11]
```diff
@@ -126,23 +123,22 @@
 # Database
 # https://docs.djangoproject.com/en/3.1/ref/settings/#databases
 
+DJANGO_DATABASE_NAME = os.environ.get("DJANGO_DATABASE_NAME", 'default')
+DJANGO_DATABASE_ENGINE = os.environ.get("DJANGO_DATABASE_ENGINE", 'django.db.backends.mysql')
```
Comment (Sispheor) [inline Squest/settings.py:127]: doc

File: Squest/views.py [hunk 12]
```diff
@@ -25,10 +25,10 @@ def home(request):
         context['total_request_need_info'] = all_requests.filter(state=RequestState.NEED_INFO).count()
 
     if request.user.has_perm('service_catalog.list_instance'):
-        context['total_instance'] = all_instances.filter(state='AVAILABLE').count()
+        context['total_instance'] = all_instances.filter(state=InstanceState.AVAILABLE).count()
 
     if request.user.has_perm('service_catalog.list_support'):
-        context['total_support_opened'] = all_supports.filter(state='OPENED').count()
+        context['total_support_opened'] = all_supports.filter(state=SupportState.OPENED).count()
```
Comment (Sispheor) [inline Squest/views.py:31]: remove admin_data.py

File: docs/manual/administration/extras.md [hunk 13]
```diff
@@ -1,15 +1,14 @@
...
+Instance/Requests hooks are a way to call a RHAAP/AWX job template following the new state of a `Request` or an `Instance`.
 
 For example, if you want to call a job template that performs an action everytime a `Request` switch to `FAILED` state.
 
-Form field:
+Create a RequestHook:
 
 - **name:** Name of your hook
-- **Model:** Target model object that will be linked to the hook (`Request` or an `Instance`)
 - **State:** State of the selected `model`. The hook will be triggered when an instance of the select model type will switch to this selected state
```
Comment (Sispheor) [inline docs/manual/administration/extras.md:12]: to be reviews as model doesn't exist anymore

File: profiles/templatetags/squest_utils.py [hunk 14]
```diff
@@ -248,11 +248,19 @@ def generate_sidebar(user):
...
-                        'permission_required': 'service_catalog.list_globalhook',
-                        'active': ["globalhook_list", "globalhook_create", "globalhook_edit", "globalhook_delete"]
+                        'permission_required': 'service_catalog.list_requesthook',
+                        'active': ["requesthook_list", "requesthook_create", "requesthook_edit",
+                                   "requesthook_delete"]
+                    },
+                    {
+                        'name': 'Instance hook ',
+                        'view_name': 'service_catalog:instancehook_list',
+                        'icon': 'fas fa-file-code',
```
Comment (Sispheor) [inline profiles/templatetags/squest_utils.py:261]: far

File: service_catalog/migrations/0023_auto_20230912_1524.py [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('service_catalog', '0022_auto_20230906_1539'),
+    ]
+
+    operations = [
+
+        migrations.RunSQL('UPDATE service_catalog_instance SET state = 1 WHERE state = "SUBMITTED";'),
+        migrations.RunSQL('UPDATE service_catalog_instance SET state = 1 WHERE state = "PENDING";'),
```
Comment (Sispheor) [inline service_catalog/migrations/0023_auto_20230912_1524.py:15]: should be 2

File: tests/test_service_catalog/test_views/test_admin/test_tools/test_request_hook/test_edit.py [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+        self.url = reverse('service_catalog:requesthook_edit', kwargs=args)
+
+    def test_get_page(self):
+        response = self.client.get(self.url)
+        self.assertEqual(200, response.status_code)
+
+    def test_edit_request_hook(self):
+        data = {
+            "name": "hook2",
+            "model": "Request",
```
Comment (Sispheor) [inline tests/test_service_catalog/test_views/test_admin/test_tools/test_request_hook/test_edit.py:26]: model

**Review by Sispheor — APPROVED** [review]

**Pull request merged**